src/Features/CSharp/Portable/MoveToResx/CSharpMoveToResxCodeFixProvider.cs

```diff
@@ -0,0 +1,474 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.MoveToResx;
+
+[ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.MoveToResx), Shared]
+internal sealed class CSharpMoveToResxCodeFixProvider : CodeFixProvider
+{
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public CSharpMoveToResxCodeFixProvider() { }
+
+    public override ImmutableArray<string> FixableDiagnosticIds
+        => [IDEDiagnosticIds.MoveToResxDiagnosticId];
+
+    public override FixAllProvider GetFixAllProvider()
+        => new CSharpMoveToResxFixAllProvider();
+
+    public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        var diagnostic = context.Diagnostics[0];
+        var root = await context.Document.GetRequiredSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+        var node = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true) as LiteralExpressionSyntax;
+        if (node is null)
+            return;
+
+        var allResx = GetAllResxFiles(context.Document.Project).ToList();
+        if (allResx.Count == 0)
+        {
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    title: CSharpFeaturesResources.No_resx_file_found_Please_create_a_resx_file_in_your_project_or_folder,
+                    createChangedDocument: c => Task.FromResult(context.Document), // No-op
+                    equivalenceKey: "NoResxFileFound"),
+                diagnostic);
```

Comment:
i don't love this.  why don't we make them a resx file on their behalf?

This is also problematic as this basically shows a squiggle *and* a lightbulb.  but hte latter does nothing to fix the squiggle.